opentelemetry-sdk/src/opentelemetry/sdk/_configuration/__init__.py

```diff
@@ -354,37 +354,61 @@ def _import_id_generator(id_generator_name: str) -> IdGenerator:
     raise RuntimeError(f"{id_generator_name} is not an IdGenerator")
 
 
-def _initialize_components(auto_instrumentation_version):
-    trace_exporters, metric_exporters, log_exporters = _import_exporters(
-        _get_exporter_names("traces"),
-        _get_exporter_names("metrics"),
-        _get_exporter_names("logs"),
+def _initialize_components(
+    auto_instrumentation_version: Optional[str] = None,
+    span_exporter_names: Optional[List[str]] = None,
+    metric_exporter_names: Optional[List[str]] = None,
+    log_exporter_names: Optional[List[str]] = None,
+    sampler: Optional[Sampler] = None,
+    resource_attributes: Optional[Attributes] = None,
+    id_generator: IdGenerator = None,
+    logging_enabled: Optional[bool] = None,
+):
+    if span_exporter_names is None:
+        span_exporter_names = []
+    if metric_exporter_names is None:
+        metric_exporter_names = []
+    if log_exporter_names is None:
+        log_exporter_names = []
+    span_exporters, metric_exporters, log_exporters = _import_exporters(
+        span_exporter_names + _get_exporter_names("traces"),
```

Comment:
Should this be additive or should we have the in-code configuration take priority and replace entrypoints similar to how we override env var? Would users be surprised by the "default" behavior?

Reply:
I think both is better since it allows users to add without code changes even if they already are passing in. Furthermore, the point of this is mainly to enable custom configurators to add exporters, we should leave open the options of distros allowing for additional env var customization. And if a configurator really wants to block that option, they could delete the env var